src/map/battle.c

```diff
@@ -4047,7 +4047,8 @@ struct Damage battle_calc_misc_attack(struct block_list *src,struct block_list *
 #endif
 			md.damage = ( (sd?pc->checkskill(sd,NC_MAINFRAME):10) + 8 ) * ( skill_lv + 1 ) * ( status_get_sp(src) + sstatus->vit );
 			RE_LVL_MDMOD(100);
-			md.damage += status_get_hp(src) - totaldef;
+
+			md.damage += ((status_get_hp(src) < totaldef) ? totaldef : status_get_hp(src)) - totaldef;
```

Comment:
Split this into another commit?
Since this is not with item type, but some underflow check...